Jenkinsfile

```diff
@@ -50,6 +50,103 @@ for (int i = 0; i < platforms.size(); ++i) {
         }
     }
 }
-
-/* Execute our platforms in parallel */
 parallel(branches)
+
+// Integration testing, using a locally built Docker image
+def itBranches = [:]
```

Comment:
There is definitely some level of copy paste below. I think we'll want to factor this out at some point. 

But I didn't want to spend too much making this generic yet (like having a way to simply provide the command the pass, the expected result, etc.). I'd rather first have this merged as-is, once green, and we iterate on growing the test coverage and will refactor in followups.